File: ios/NorthAx/Networking/DTOs/AuthDTOs.swift
import Foundation

// Auth DTOs mirroring the OpenAPI schemas (§6.1). Property names match the
// camelCase wire keys 1:1, so no CodingKeys are needed.

struct EmailSignInRequest: Encodable {
    var email: String
    var password: String
}

struct EmailSignUpRequest: Encodable {
    var name: String
    var email: String
    var password: String
}

struct UserSummaryDTO: Decodable {
    var id: String
    var name: String
    var email: String?
}

struct AuthResponse: Decodable {
    var accessToken: String
    var refreshToken: String
    var user: UserSummaryDTO
}

struct RefreshRequest: Encodable {
    var refreshToken: String
}

struct RefreshResponse: Decodable {
    var accessToken: String
    var refreshToken: String
}

struct UserProfileDTO: Decodable {
    var id: String
    var name: String
    var email: String?
    var createdAt: Date
}

struct UpdateProfileRequest: Encodable {
    var name: String
}

File: android/app/src/main/java/app/northax/data/remote/dto/AuthDtos.kt
package app.northax.data.remote.dto

import app.northax.data.remote.ApiInstant
import kotlinx.serialization.Serializable

// Auth DTOs mirroring the OpenAPI schemas. Property names match the camelCase
// wire keys 1:1.

@Serializable
data class EmailSignInRequest(val email: String, val password: String)

@Serializable
data class EmailSignUpRequest(val name: String, val email: String, val password: String)

@Serializable
data class UserSummaryDto(val id: String, val name: String, val email: String? = null)

@Serializable
data class AuthResponse(
    val accessToken: String,
    val refreshToken: String,
    val user: UserSummaryDto,
)

@Serializable
data class RefreshRequest(val refreshToken: String)

@Serializable
data class RefreshResponse(val accessToken: String, val refreshToken: String)

@Serializable
data class UserProfileDto(
    val id: String,
    val name: String,
    val email: String? = null,
    val createdAt: ApiInstant,
)

@Serializable
data class UpdateProfileRequest(val name: String)
